linera-views-derive/src/lib.rs

```diff
@@ -37,33 +37,50 @@ impl<'a> Constraints<'a> {
     }
 }
 
-fn get_extended_entry(e: Type) -> TokenStream2 {
-    let syn::Type::Path(typepath) = e else {
-        panic!("The type should be a path");
+fn get_extended_entry(e: Type) -> Result<TokenStream2, Error> {
+    let syn::Type::Path(typepath) = &e else {
+        return Err(Error::new_spanned(e, "Expected a path type"));
     };
-    let path_segment = typepath.path.segments.into_iter().next().unwrap();
-    let ident = path_segment.ident;
-    let arguments = path_segment.arguments;
-    quote! { #ident :: #arguments }
+    let path_segment = match typepath.path.segments.first() {
+        Some(segment) => segment,
+        None => return Err(Error::new_spanned(&typepath.path, "Path has no segments")),
+    };
+    let ident = &path_segment.ident;
+    let arguments = &path_segment.arguments;
+    Ok(quote! { #ident :: #arguments })
 }
 
-fn generate_view_code(input: ItemStruct, root: bool) -> TokenStream2 {
+fn generate_view_code(input: ItemStruct, root: bool) -> Result<TokenStream2, Error> {
+    // Validate that all fields are named
+    for field in &input.fields {
+        if field.ident.is_none() {
+            return Err(Error::new_spanned(field, "All fields must be named."));
+        }
+    }
+
     let Constraints {
         input_constraints,
         impl_generics,
         type_generics,
     } = Constraints::get(&input);
 
-    let attrs: StructAttrs = deluxe::parse_attributes(&input).unwrap();
-    let context = attrs.context.unwrap_or_else(|| {
-        let ident = &input
-            .generics
-            .type_params()
-            .next()
-            .expect("no `context` given and no type parameters")
-            .ident;
-        parse_quote! { #ident }
-    });
+    let attrs: StructAttrs = deluxe::parse_attributes(&input)
+        .map_err(|e| Error::new_spanned(&input, format!("Failed to parse attributes: {}", e)))?;
+    let context = match attrs.context {
+        Some(ctx) => ctx,
+        None => match input.generics.type_params().next() {
+            Some(param) => {
```

Comment:
I think this could be more clearly expressed using [`or_else`](https://doc.rust-lang.org/std/option/enum.Option.html#method.or_else).